streaming/base/stream.py

```diff
@@ -306,7 +306,11 @@ def get_shards(self, world: World) -> List[Reader]:
         wait_for_file_to_exist(filename, TICK, self.download_timeout,
                                f'{filename} file took too long to download')
 
-        obj = json.load(open(filename))
+        try:
+            obj = json.load(open(filename))
+        except json.decoder.JSONDecodeError as error:
+            error.args = (f'Either {filename} is empty or corrupted. ' + error.args[0],)
```

Comment:
Index file at {filename} is empty or corrupted?

Reply:
Done in https://github.com/mosaicml/streaming/pull/242/commits/004a43a9396d6b9b7a673015ca550d7d9cb2bffd.